File: Yona/Yona/Core/AppConfig.swift
//
//  AppConfig.swift
//  Yona
//
//  Reads configuration from a bundled, git-ignored `Supabase.plist` (template:
//  Supabase.example.plist at the repo root). The file ships inside the app but
//  is never committed — RLS guards the data; the Brandfetch ID is public-by-design.
//

import Foundation

struct AppConfig {
    let supabaseURL: URL
    let supabaseAnonKey: String

    /// `false` until a real `Supabase.plist` is present.
    let isConfigured: Bool

    /// Brandfetch Logo API client ID; nil when unset (logos fall back to letter-tiles).
    let brandfetchClientID: String?

    static let shared = AppConfig()

    init(bundle: Bundle = .main) {
        var urlString = ""
        var key = ""
        var brandfetch: String?

        if let plistURL = bundle.url(forResource: "Supabase", withExtension: "plist"),
           let dict = NSDictionary(contentsOf: plistURL) {
            urlString = (dict["SUPABASE_URL"] as? String) ?? ""
            key = (dict["SUPABASE_ANON_KEY"] as? String) ?? ""
            let bf = ((dict["BRANDFETCH_CLIENT_ID"] as? String) ?? "")
                .trimmingCharacters(in: .whitespacesAndNewlines)
            brandfetch = (bf.isEmpty || bf.contains("YOUR-")) ? nil : bf
        }

        urlString = urlString.trimmingCharacters(in: .whitespacesAndNewlines)
        key = key.trimmingCharacters(in: .whitespacesAndNewlines)

        if !urlString.isEmpty, !key.isEmpty,
           !urlString.contains("YOUR-REF"),
           let url = URL(string: urlString), url.host != nil {
            supabaseURL = url
            supabaseAnonKey = key
            isConfigured = true
        } else {
            supabaseURL = URL(string: "https://placeholder.supabase.co")!
            supabaseAnonKey = "placeholder-anon-key"
            isConfigured = false
        }

        brandfetchClientID = brandfetch
    }
}

File: android/app/src/main/java/com/yona/app/core/AppConfig.kt
package com.yona.app.core

import com.yona.app.BuildConfig

/**
 * App configuration backed by generated [BuildConfig] fields, which are populated
 * at build time from the git-ignored `secrets.properties` (see the app build script).
 * Mirrors the iOS app's `AppConfig` reading from the bundled `Supabase.plist`.
 */
object AppConfig {
    val supabaseUrl: String = BuildConfig.SUPABASE_URL
    val supabaseAnonKey: String = BuildConfig.SUPABASE_ANON_KEY
    val brandfetchClientId: String = BuildConfig.BRANDFETCH_CLIENT_ID

    /** True only when real Supabase credentials are present (not a placeholder/empty). */
    val isConfigured: Boolean
        get() = supabaseUrl.isNotBlank() &&
            supabaseAnonKey.isNotBlank() &&
            !supabaseUrl.contains("YOUR-PROJECT", ignoreCase = true)

    /** Host portion of the Supabase URL, for display (e.g. `rxjw….supabase.co`). */
    val supabaseHost: String
        get() = supabaseUrl
            .removePrefix("https://")
            .removePrefix("http://")
            .substringBefore("/")
}
